tests/Requests/RequestTest.php

```diff
@@ -5,220 +5,92 @@
 namespace OpenEuropa\EPoetry\Tests\Requests;
 
 use GuzzleHttp\Psr7\Response;
-use OpenEuropa\EPoetry\Tests\AbstractTest;
-use OpenEuropa\EPoetry\Type\ContactPersonIn;
-use OpenEuropa\EPoetry\Type\Contacts;
+use OpenEuropa\EPoetry\Serializer\RequestsSerializer;
 use OpenEuropa\EPoetry\Type\CreateRequests;
-use OpenEuropa\EPoetry\Type\CreateRequestsResponse;
-use OpenEuropa\EPoetry\Type\LinguisticRequestIn;
-use OpenEuropa\EPoetry\Type\RequestGeneralInfoIn;
 
 /**
  * @internal
  * @coversNothing
  */
-final class RequestTest extends AbstractTest
+final class RequestTest extends AbstractRequestTest
 {
     /**
-     * Test a SOAP request.
+     * Data provider.
+     *
+     * @return mixed
      */
-    public function testRequestSending()
+    public function requestSendingCases()
     {
-        // Generate General Info.
-        $generalInfo = new RequestGeneralInfoIn();
-
-        $generalInfo->setTitle('Test')
-            ->setInternalReference('1')
-            ->setInternalTechnicalId('1')
-            ->setRequestedDeadline(new \DateTime('2020-02-02'))
-            ->setSensitive(false)
-            ->setDocumentToBeAdopted(true)
-            ->setDecideReference('decideReference')
-            ->setSentViaRUE(true)
-            ->setDestinationCode('PUBLIC')
-            ->setSlaAnnex('')
-            ->setSlaCommitment('')
-            ->setComment('')
-            ->setOnBehalfOf('')
-            ->setAccessibleTo('');
-
-        // Generate Contacts.
-        $contacts = new Contacts();
-
-        $contact = new ContactPersonIn();
-        $contact->setUserId('1');
-        $contact->setRoleCode('AUTHOR');
-        $contacts->addContact($contact);
-
-        $contact = new ContactPersonIn();
-        $contact->setUserId('2');
-        $contact->setRoleCode('EDITOR');
-        $contacts->addContact($contact);
+        return $this->getFixture('request-send-test.yml');
+    }
 
-        $linguisticRequestIn = new LinguisticRequestIn();
-        $linguisticRequestIn->setContacts($contacts)
-            ->setGeneralInfo($generalInfo);
+    /**
+     * Data provider.
+     *
+     * @return mixed
+     */
+    public function responseParsingCases()
+    {
+        return $this->getFixture('response-parse-test.yml');
+    }
 
-        $createRequests = new CreateRequests();
-        $createRequests->setLinguisticRequest([$linguisticRequestIn]);
+    /**
+     * Test a SOAP request.
+     *
+     * @dataProvider requestSendingCases
+     *
```

Comment:
How do you see that ?
It wasn't there before!
It's not that I don't trust you, but I would like to be able to spot issues like that as well when doing review! I'm curious :)

Reply:
In a test there is no point of documenting what exceptions are going to be thrown.

Reply:
@ademarco PHPStorm adds them automatically. On my side it wasn't added because for some reason I had disabled the feature.
That said, there or not, I don't think it's problem.
Don't you ?

Reply:
Yes, it's ugly to see, PHPStorm underlines the all doc block because this missing missing.